### PR TITLE
Improved async call to okhttp. 

### PR DESCRIPTION
**Pull request by ji11er**
# Description

Former solution just wrapped a blocking call, now we using asynch call and collect the results via callbacks. Mono's are created based on the result. This make more efficient use of the resources since nothing is blocking now. 

Reformatted according new checkstyle

## Issue reference
#19 
#46 

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by artursouza** [comment]
The Java SDK is WIP, so we are not merging into master yet. I changed the base of this PR to the right branch. One we are done with this milestone, we should be able to use master directly again.

File: sdk/src/main/java/io/dapr/actors/AbstractClientBuilder.java [hunk 1]
```diff
@@ -2,48 +2,46 @@
...
-     * @param port New port.
-     * @return This instance.
-     */
-    public AbstractClientBuilder withPort(int port) {
-        this.port = port;
-        return this;
+  /**
+   * Default port for Dapr after checking environment variable.
+   */
+  private int port = AbstractClientBuilder.GetEnvPortOrDefault();
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/actors/AbstractClientBuilder.java]: Any particular reason to move from 4 spaces to 2 spaces? I am good either way but sticking to 4 spaces would reduce the diff :)

File: sdk/src/main/java/io/dapr/actors/AbstractDaprClient.java [hunk 2]
```diff
@@ -1,119 +1,173 @@
...
+  }
+
+  public interface DaprHttpCallback {
 
     /**
-     * Invokes an API asynchronously that returns a text payload.
-     * @param method HTTP method.
-     * @param urlString url as String.
-     * @param json JSON payload or null.
-     * @return Asynchronous text
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/actors/AbstractDaprClient.java]: Why is the return comment removed?

File: sdk/src/main/java/io/dapr/actors/AbstractDaprClient.java [hunk 3]
```diff
@@ -1,119 +1,173 @@
...
+    String requestId = UUID.randomUUID().toString();
+    RequestBody body = json != null ? RequestBody.create(MEDIA_TYPE_APPLICATION_JSON, json) : REQUEST_BODY_EMPTY_JSON;
+
+    Request request = new Request.Builder()
+        .url(new URL(this.baseUrl + urlString))
+        .method(method, body)
+        .addHeader(Constants.HEADER_DAPR_REQUEST_ID, requestId)
+        .build();
+
+    this.httpClient.newCall(request).enqueue(new Callback() {
```
Comment (artursouza) [inline sdk/src/main/java/io/dapr/actors/AbstractDaprClient.java]: Thanks for that.

File: sdk/src/main/java/io/dapr/actors/AbstractClientBuilder.java (hunk 1 above)
Comment (ji11er) [reply sdk/src/main/java/io/dapr/actors/AbstractClientBuilder.java]: I'd like to preserve horizontal space. If you can agree, I'd prefer. Unfortunately, I had two commits one with small changes and one more with the formatting only. But I created a patch to follow the rules for pull requests and patched my forked copy of the repo. Therefore the commit is huge.

File: sdk/src/main/java/io/dapr/actors/AbstractDaprClient.java (hunk 3 above)
Comment (ji11er) [reply sdk/src/main/java/io/dapr/actors/AbstractDaprClient.java]: Never mind

**Comment by ji11er** [comment]
I promise less headache with upcoming commits.

**Comment by ji11er** [comment]
I hit that button accidentially before the review was finished

**Pull request merged**